examples/readgpio.py

```diff
@@ -0,0 +1,71 @@
+"""
+Very basic example showing how to read a GPIO pin from the ULP and access
+that data from the main CPU.
+
+In this case GPIO4 is being read. Note that the ULP needs to refer to GPIOs
+via their RTC channel number. You can see the mapping in this file:
+https://github.com/espressif/esp-idf/blob/1cb31e50943bb757966ca91ed7f4852692a5b0ed/components/soc/esp32/include/soc/rtc_io_channel.h#L51
```

Comment:
do they have usable git tags in their repo?

Reply:
Did you mean to have a shorter url here or maybe a more friendly url?

They do tag their releases, so this link also works:
https://github.com/espressif/esp-idf/blob/v4.4.1/components/soc/esp32/rtc_io_periph.c

I can change to those - I believe MicroPython on the ESP32 is currently built with ESP-IDF v4.x and v4.4.1 seems to be the latest release before v5.

Reply:
A link to a release tag is definitely preferable than one to a random changeset.

Reply:
Ok. I updated it to v4.4.1.